source/common/upstream/cluster_manager_impl.cc

```diff
@@ -356,7 +347,8 @@ void ClusterManagerImpl::onClusterInit(Cluster& cluster) {
     //
     // See https://github.com/envoyproxy/envoy/pull/3941 for more context.
     bool scheduled = false;
-    const auto merge_timeout = mergeTimeout(cluster);
+    const auto merge_timeout =
+        PROTOBUF_GET_MS_OR_DEFAULT(cluster.info()->lbConfig(), update_merge_window, 1000);
```

Comment:
TBH I don't think it matters since this is already operating on the stored proto object, but will wait to see if anyone else feels differently before merging.